WebApp-B2C-DotNet/App_Start/Startup.Auth.cs

```diff
@@ -44,59 +38,54 @@ public void ConfigureAuth(IAppBuilder app)
 
             app.UseCookieAuthentication(new CookieAuthenticationOptions());
 
-            OpenIdConnectAuthenticationOptions options = new OpenIdConnectAuthenticationOptions
+            // Configure OpenID Connect middleware for each policy
+            app.UseOpenIdConnectAuthentication(CreateOptionsFromPolicy(SignUpPolicyId));
+            app.UseOpenIdConnectAuthentication(CreateOptionsFromPolicy(ProfilePolicyId));
+            app.UseOpenIdConnectAuthentication(CreateOptionsFromPolicy(SignInPolicyId));
+        }
+
+        // Used for avoiding yellow-screen-of-death
+        private Task AuthenticationFailed(AuthenticationFailedNotification<OpenIdConnectMessage, OpenIdConnectAuthenticationOptions> notification)
+        {
+            notification.HandleResponse();
+            if (notification.Exception.Message == "access_denied")
+            {
+                notification.Response.Redirect("/");
```

Comment:
Yes for these B2C flows `access_denied` is only returned when the user clicks "Cancel" on the B2C page, so I was redirecting to the home page.  A better experience would probably be to redirect to the page the user was on previously.  But I don't think OWIN exposes the `redirect_uri` from the OAuth state in this particular notification, so I'm not sure that's possible without adding a bunch of code for keeping track in cookies.